### PR TITLE
[acr] support anonymous pull

### PR DESCRIPTION
**Pull request by yugangw-msft**
~Label "Do Not Merge" as this is pending SDK release~
edit: azure-mgmt-containerregistry 3.0.0rc17 was released

**Description**
Support registry-wide pull from unauthenticated clients

**Testing Guide**
az acr update -n myRegistry --anonymous-pull-enabled

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
acr

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 1]
```diff
@@ -94,6 +94,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
     with self.argument_context('acr update', arg_group='Network Rule') as c:
         c.argument('data_endpoint_enabled', get_three_state_flag(), help="Enable dedicated data endpoint for client firewall configuration")
 
+    with self.argument_context('acr update') as c:
+        c.argument('anonymous_pull_enabled', get_three_state_flag(), help="Enables registry-wide pull from unauthenticated clients")
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:98]:
```suggestion
        c.argument('anonymous_pull_enabled', get_three_state_flag(), help="Enable registry-wide pull from unauthenticated clients")
```

**Review by yungezz — APPROVED** [review]

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 2]
```diff
@@ -29,7 +29,7 @@ azure-mgmt-cognitiveservices==6.3.0
 azure-mgmt-compute==19.0.0
 azure-mgmt-consumption==2.0.0
 azure-mgmt-containerinstance==1.5.0
-azure-mgmt-containerregistry==3.0.0rc16
+azure-mgmt-containerregistry==3.0.0rc17
```
Comment (yungezz) [inline src/azure-cli/requirements.py3.Darwin.txt:32]: since SDK version changed, I'll suppose you've already run live tests :)

Comment (yugangw-msft) [reply src/azure-cli/requirements.py3.Darwin.txt:32]: @yungezz There is no api version bump in this version. The spec change PR is here FYI: https://github.com/Azure/azure-rest-api-specs/pull/13001
But I still ran a few more tests live to provide enough confidence for this PR. Please take a look and let me know if you have any other suggestions.

**Review by yungezz — APPROVED** [review]

**Pull request merged**